global_helpers/panther_base_helpers.py

```diff
@@ -215,19 +215,26 @@ def okta_alert_context(event: dict):
 
 
 def crowdstrike_detection_alert_context(event: dict):
-    """Returns common context for Crowstrike detections"""
+    """Returns common context for Crowdstrike detections"""
     return {
-        "user": event.get("UserName", ""),
-        "console-link": event.get("FalconHostLink", ""),
-        "commandline": event.get("CommandLine", ""),
-        "parentcommandline": event.get("ParentCommandLine", ""),
-        "filename": event.get("FileName", ""),
-        "filepath": event.get("FilePath", ""),
-        "description": event.get("DetectDescription", ""),
-        "action": event.get("PatternDispositionDescription", ""),
+        "user": get_crowdstrike_field(event, "UserName", default=""),
+        "console-link": get_crowdstrike_field(event, "FalconHostLink", default=""),
+        "commandline": get_crowdstrike_field(event, "CommandLine", default=""),
+        "parentcommandline": get_crowdstrike_field(event, "ParentCommandLine", default=""),
+        "filename": get_crowdstrike_field(event, "FileName", default=""),
+        "filepath": get_crowdstrike_field(event, "FilePath", default=""),
+        "description": get_crowdstrike_field(event, "DetectDescription", default=""),
+        "action": get_crowdstrike_field(event, "PatternDispositionDescription", default=""),
     }
 
 
+def get_crowdstrike_field(event, field, default=None):
```

Comment:
```suggestion
def get_crowdstrike_field(event, name, default=None):
```

You can also consider adding a comment about the precedence/search order.

Reply:
`field_name` maybe?

Reply:
Sure!